src/pybind/mgr/ansible/module.py

```diff
@@ -179,18 +186,77 @@ def is_effective(self):
         In the case of Ansible, this will be True if the playbooks has been
         executed succesfully.
 
-        :return Boolean: if the playbook has been executed succesfully
+        :return Boolean: if the playbook/ARS operation has been executed
+                         succesfully
         """
 
-        return self.status == ExecutionStatusCode.SUCCESS
+        return self._status == ExecutionStatusCode.SUCCESS
 
     @property
     def is_errored(self):
-        return self.error
+        return self._status == ExecutionStatusCode.ERROR
 
     @property
-    def is_complete(self):
-        return self.is_errored or (self.is_persistent and self.is_effective)
+    def result(self):
+        return self._result
+
+
+class ARSChangeOperation(AnsibleChangeOperation):
+    """Execute an Ansible Runner Service Operation that implies a change in
+    the cluster
+    """
+    def __init__(self, client, logger, url, http_operation, payload={}):
```

Comment:
FYI, `payload={}` is a [mutable default argument](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments).

Reply:
thx!. Fixed.